com.android.builder.internal.aapt.v2.Aapt2Exception: Android resource linking failed

## Issue

Android:  gradle release build fails in verifyReleaseResources step.

Stack trace : 
```
> Task :react-native-app-auth:verifyReleaseResources FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-app-auth:verifyReleaseResources'.
> java.util.concurrent.ExecutionException: com.android.builder.internal.aapt.v2.Aapt2Exception: Android resource linking failed
  Output:  /Users/Devs/trash/ReactNativeTestAppAuth/node_modules/react-native-app-auth/android/build/intermediates/res/merged/release/values-v28/values-v28.xml:7: error: resource android:attr/dialogCornerRadius not found.
  /Users/Devs/trash/ReactNativeTestAppAuth/node_modules/react-native-app-auth/android/build/intermediates/res/merged/release/values-v28/values-v28.xml:11: error: resource android:attr/dialogCornerRadius not found.
  /Users/Devs/trash/ReactNativeTestAppAuth/node_modules/react-native-app-auth/android/build/intermediates/res/merged/release/values/values.xml:961: error: resource android:attr/fontVariationSettings not found.
  /Users/Devs/trash/ReactNativeTestAppAuth/node_modules/react-native-app-auth/android/build/intermediates/res/merged/release/values/values.xml:962: error: resource android:attr/ttcIndex not found.
  error: failed linking references.
```

Test repository : https://github.com/rmkpatchaa/ReactNativeTestAppAuth

How to reproduce : 
```
npm install
cd android
./gradlew assembleRelease
```

---

## Environment

<!-- change `[ ]` to `[x]` to select an option(s) -->
- **Your Identity Provider**: `e.g. IdentityServer 4 / Okta / Azure`
- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [x] **Android** but have not tested behavior on iOS
  - [ ] Both
- **Are you using Expo, e.g. `ExpoKit`?**
  - [x] No
  - [ ] Yes, I've _not_ ejected
  - [ ] Yes, but I **have** ejected to `ExpoKit`
  - [ ] Yes, but I **have** ejected to vanilla React Native

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -41,6 +41,6 @@ repositories {
 }
 
 dependencies {
-    implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
-    implementation "net.openid:appauth:0.7.1"
+    compile 'com.facebook.react:react-native:'+safeExtGet('reactNativeVersion', '+')
+    compile 'net.openid:appauth:0.7.1'
 }

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,23 +1,30 @@
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
 
 buildscript {
     repositories {
+        maven {
+            url 'https://maven.google.com/'
+            name 'Google'
+        }
         jcenter()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.3.2'
+        classpath 'com.android.tools.build:gradle:3.2.1'
     }
 }
 
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 27
-    buildToolsVersion '28.0.2'
+    compileSdkVersion safeExtGet('compileSdkVersion', 27)
+    buildToolsVersion safeExtGet('buildToolsVersion', '27.0.2')
 
     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 26
+        minSdkVersion safeExtGet('minSdkVersion', 16)
+        targetSdkVersion safeExtGet('targetSdkVersion', 26)
         versionCode 1
         versionName "1.0"
         manifestPlaceholders = [
@@ -34,6 +41,6 @@ repositories {
 }
 
 dependencies {
-    implementation 'com.facebook.react:react-native:+'
+    implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
     implementation "net.openid:appauth:0.7.1"
 }

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -20,7 +20,7 @@ apply plugin: 'com.android.library'
 
 android {
     compileSdkVersion safeExtGet('compileSdkVersion', 27)
-    buildToolsVersion safeExtGet('buildToolsVersion', '27.0.2')
+    buildToolsVersion safeExtGet('buildToolsVersion', '28.0.2')
 
     defaultConfig {
         minSdkVersion safeExtGet('minSdkVersion', 16)

